src/block_disagg/block_disagg_write.c

```
@@ -286,8 +282,15 @@ __wti_block_disagg_page_discard(
       ", reconciliation_id %" PRIu64 ", size %" PRIu32 ", checksum %" PRIx32,
       page_id, lsn, checkpoint_id, reconciliation_id, size, checksum);
 
-    /* FIXME-WT-14532: Implement the page discard logic. */
-    WT_UNUSED(block_disagg);
-
-    return (0);
+    /* Send the discard request. */
+    WT_PAGE_LOG_HANDLE *plhandle = block_disagg->plhandle;
+    WT_PAGE_LOG_DISCARD_ARGS discard_args;
+    WT_CLEAR(discard_args);
+    discard_args.backlink_lsn = block_meta->backlink_lsn;
+    discard_args.base_lsn = block_meta->base_lsn;
+    discard_args.backlink_checkpoint_id = block_meta->backlink_checkpoint_id;
+    discard_args.base_checkpoint_id = block_meta->base_checkpoint_id;
+
+    return (
```

Until mongod implements this call, please check this function pointer for NULL, and maybe just silently ignore it?

---

Sure, see a4ac21e7a1e7f07c68ebeb480a70007d2b0055cc. Should we have a FIXME to fully enable it once implemented on their side?

---

Yes, feel free to add it. This should be a required call for all PALI implementations.

---

Added in 97b4634d4a26be64f5630d9a8bc8a5d4d51ce78f. We will eventually remove it someday since it's not there for put/get and it will be a requirement.